contracts/contracts/coordination/subscription/StandardSubscription.sol

```diff
@@ -242,20 +288,36 @@ contract StandardSubscription is EncryptorSlotsSubscription, Initializable, Owna
         uint256 fees = encryptorFees(additionalEncryptorSlots, duration);
         billing.encryptorSlots += additionalEncryptorSlots;
 
-        feeToken.safeTransferFrom(msg.sender, address(this), fees);
+        paidFees += fees;
         emit EncryptorSlotsPaid(msg.sender, fees, additionalEncryptorSlots, endOfCurrentPeriod);
+        return fees;
     }
 
     /**
-     * @notice Withdraws the contract balance to the treasury
+     * @notice Withdraws the fees to the treasury
      * @param amount The amount to withdraw
      */
     function withdrawToTreasury(uint256 amount) external onlyOwner {
-        require(amount <= feeToken.balanceOf(address(this)), "Insufficient balance available");
+        require(amount <= paidFees, "Insufficient balance available");
+        paidFees -= amount;
         feeToken.safeTransfer(msg.sender, amount);
         emit WithdrawalToTreasury(msg.sender, amount);
     }
 
+    /**
+     * @notice Withdraws unused fees to the adopter setter
+     * @param amount The amount to withdraw
+     */
+    function refund(uint256 amount) external {
+        require(msg.sender == adopterSetter, "Only adopter setter can call refund");
+        require(
+            feeToken.balanceOf(address(this)) - paidFees >= amount,
+            "Insufficient balance available"
+        );
+        feeToken.safeTransfer(msg.sender, amount);
+        emit Refund(msg.sender, amount);
+    }
```

Comment:
Is it possible to withdraw fees such that the total number of encryptor slots decreases at any time? Because in our current advertised rules (in our invoices etc), this is stated as only possible at renewal time (i.e. by paying less when renewing, such that the number of slots in the next period is lower). In other words, once you pay for an encryptor slot for one year, there's no way to get that money back. 

 It's ok for now if slot reduction + refunds can happen at any time, but want to clarify this and explain this rule correctly to prospective adopters.